### ISSUE TITLE
Problems when try to watch 2 channels of the same transponder

### ISSUE THREAD

**Issue by roisiano** [Jalle19/satip-axe#39]
Hello:

With version 23:
https://github.com/Jalle19/satip-axe/releases/tag/build23 

It seems that system "becomes crazy" when I try to see 2 channels os the same TP, and I can't see no one of them until I stop one of the 2 channels of the same TP. 

MINISATIP="yes"
MINISATIP_OPTS="-j 0:0-1210,2:1-1420 -Z 0,2:600 --dmx-source 0:0,1:1,2:0,3:3 --diseqc 1:4-0,3:4-0"

I tried with channels from same TP with unicable. NOT working with version 23.

**When I returned to version 22, different channels from same transponder (TP) working again.**  Same configuration in both cases.

By the way, I'm not sure that upgrade-fw option is working well, at least for "downgrades" as it has been the case. 

I live far from the site where I have the Digibit R1, and tried to make a remote:
upgrade-fw satip-axe-202208090718-22.fw
 
from the folder where the archive was.

I'm not sure about what happened, but after trying it and rebooting, the device only started from original firmware (I use minisatip from USB stick), and, when I was at the site where I have the Digibit R1, some files were missing from the USB stick. 
After using "new" files for version 22 from USB stick, it works well, so, perhaps upgrade-fw option has corrupted the files (not sure about it, but I'me sure that it did not work after trying the "downgrade"). 

Kind regards

**Comment by Jalle19** [comment]
The `upgrade-fw` command is only for flash installations - if you're booting from a USB stick you should put the new firmware on the USB and reboot normally (I think, I've never used that mode).

Can you copy the `minisatip` binary from build 22 and upload it to a device running build 23 to see if your issue is resolved then? Then we know whether issue is in minisatip or somewhere else.

**Comment by Jalle19** [comment]
Looks like your problem could be related to the `--dmx-source` option. That option was removed in build 22 in favor of another `-A` or `--virtual-diseqc`. You're also only specifying two slots in your Unicable configuration, not sure if that's correct or not.

**Comment by roisiano** [comment]
> related to the `--dmx-source` option. That option was removed in build 22 in favor of another `-A` or `--virtual-diseqc`. You're also only specifying two slots in your Unicable configuration, not sure

Yes; but bulid 22 is working OK again with that config. NOT build 23. 

I have 2 diseqCs connected to Digibit R1. For this reason I use only 2 unicable adapters (0 and 2) with the Unicable coaxial cable connected to physical input 1. 

Not sure that the problem is the configuration. I suppose that, perhaps, minisatip ignores this: --dmx-source. When I try to watch channels from different TPs (with Unicable or diseQC), no problem at all. The problem is, with build 22, when I try to see channels from the same TP.

**Comment by Jalle19** [comment]
Try copying the minisatip binary from build 22 to build 23

**Comment by roisiano** [comment]
> minisatip binary from build 22 to build 23

Sorry, I don't understand exactly what you're proposing me to do. Rename the file " satip-axe-202210230711-23.fw" as " satip-axe-202208090718-22.fw"?

Kind regards

**Comment by Jalle19** [comment]
I'll make a custom build for you that you can try.

**Comment by Jalle19** [comment]
[satip-axe-202211161729-23-test1.zip](https://github.com/Jalle19/satip-axe/files/10024272/satip-axe-202211161729-23-test1.zip)

@roisiano can you try this build?

**Comment by roisiano** [comment]
> [satip-axe-202211161729-23-test1.zip](https://github.com/Jalle19/satip-axe/files/10024272/satip-axe-202211161729-23-test1.zip)
> 
> @roisiano can you try this build?

OK; but when I am at the place where I have the Digibit R1 (maybe within 10 or 20 days).

**Comment by Jalle19** [comment]
You can upgrade it remotely using the `upgrade-fw` script

**Comment by roisiano** [comment]
> You can upgrade it remotely using the `upgrade-fw` script

No; I use your firmware from USB. I'll check it when I'm in the place where the Digibit is.

Kind regards

**Comment by estimadarocha** [comment]
I can confirm @Jalle19 something is really wrong with this version!

Command line: minisatip -f -g -z /etc/sysconfig/minisatip -u 0:1-1420,1:0-1210,2:2-1680,3:3-2040 -b 188000 -B 14

Cesbo/astra is getting pids=all from satip-axe... and it gets crazy!

**Comment by estimadarocha** [comment]
it seens like the problem in my case is related with pids=all it doesn't work at all!

**Comment by estimadarocha** [comment]
add this log: [27/11 16:42:52.701]: tuning to 12360000(1420000) pol: h (2) sr:27500000 fec:  delsys:dvbs2 mod:  rolloff:auto pilot:auto, ts clear=2077, ts pol=2077 sequence_index = 262143
[27/11 16:42:52.701]: axe: isi -1 pls -1 mode -1
[27/11 16:42:52.709]: adapter post_tune: aid 0
[27/11 16:42:52.709]: utils.c:579 allocation_wrapper malloc allocated 4144 bytes at 0x4a9608
[27/11 16:42:52.709]: new filter 0 added for adapter 0, pid 0, flags 20, mask_len 0, master_filter 0
[27/11 16:42:52.709]: utils.c:579 allocation_wrapper malloc allocated 4144 bytes at 0x4aa640
[27/11 16:42:52.710]: new filter 1 added for adapter 0, pid 17, flags 20, mask_len 0, master_filter 1
[27/11 16:42:52.710]: Dumping pids table for adapter 0, number of unknown pids: 0
[27/11 16:42:52.710]: pid 8192, fd 0, packets 0, d/c errs 0/0, flags 2, pmt -1, filter -1, sock -1, sids: 0 -1 -1 -1 -1 -1 -1 -1
[27/11 16:42:52.711]: setting filter on PID 8192 for ADAPTER 0
[27/11 16:42:52.711]: Reply (handle 16) [127.0.0.1:60261] content_len:0, sock 11
[27/11 16:42:52.711]: MSG client << process :
HTTP/1.0 200 OK
Date: Sun, Nov 27 16:42:52 2022 GMT
Server: minisatip/1.2.38
Content-Type: video/mp2t
Connection: close


[27/11 16:42:52.744]: Flushing adapter 0 buffer of 11280 bytes after the tune 33 ms ago
[27/11 16:42:52.747]: Got the new transponder 0026 38, position 188, 35 ms after tune
[27/11 16:42:52.747]: read_dmx send=1, force_send=1, cnt 0 called for adapter 0 -> 14852 out of 188000 bytes read, 2083 ms ago (2311 228)
[27/11 16:42:52.747]: process_dmx start called for adapter 0 -> 14852 out of 188000 bytes read, 2083 ms ago
RECEIVED SIGNAL 11 - SP=49F1D0 IP=49F1D0
Obtained 1 stack frames.
sh: addr2line: not found
0x418f8a :

**Comment by Jalle19** [comment]
@estimadarocha can you test the firmware from https://github.com/Jalle19/satip-axe/issues/39#issuecomment-1317393545 ?

**Comment by estimadarocha** [comment]
@Jalle19 i used last minisatip releases .47 i think... And no problem... Your build of minisatip is broken

**Comment by Jalle19** [comment]
Did you test the build I linked?

**Comment by estimadarocha** [comment]
No... Before you asked it i already testes with own ones of minisatip GitHub. Do you want me to test yours?

**Comment by Jalle19** [comment]
It would be helpful too, yes. Good to know that 1.2.47 works though, I guess I'll make a new build using that version. Now that I think about it I have had some strange issues with 1.2.39 as well, could be related to this.

**Comment by Jalle19** [comment]
Can you both try https://github.com/Jalle19/satip-axe/releases/tag/build24-rc1 and report whether or not it works?

**Comment by estimadarocha** [comment]
In 30 mins

**Comment by estimadarocha** [comment]
sorry @Jalle19 something is really wrong

![image](https://user-images.githubusercontent.com/17991728/204159558-ff96fc8f-da4f-409d-8f56-e509afb50b09.png)

**Comment by Jalle19** [comment]
There's nothing new in this build except minisatip v1.2.47. Didn't you say that version worked when you tested earlier?

**Comment by Jalle19** [comment]
The upstream binaries are compiled with PMT=0, maybe that's the culprit.

**Comment by Jalle19** [comment]
I'd rather revert to the last working version from build 22.

**Comment by estimadarocha** [comment]
@Jalle19 can you make a build with PMT=0?

**Comment by Jalle19** [comment]
I can't seem to compile it with that option:

```
dvbapi.c: In function 'dvbapi_add_pmt':
dvbapi.c:823:17: error: 'adapter' has no member named 'transponder_id'
     k->tsid = ad->transponder_id;
                 ^
make[2]: *** [../build/dvbapi.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/build/apps/minisatip/src'
make[1]: *** [minisatip] Error 2
make[1]: Leaving directory `/build/apps/minisatip'
make: *** [apps/minisatip/minisatip] Error 2
make: *** [Makefile:102: docker-clean-release] Error 2
```

**Comment by estimadarocha** [comment]
Any idea about the cause?

**Comment by Jalle19** [comment]
```
RECEIVED SIGNAL 11 - SP=4A3120 IP=4A3120
Obtained 1 stack frames.
sh: addr2line: not found
0x4190f6 :
```
Would be helpful to get this working on these builds so we could see where it crashes

**Comment by estimadarocha** [comment]
I am fixing everything so @catalinii can have full acess to my system. By the way is there a way to have some linda of vpn packages on this machine? Openvpn or wireguard since its a low power CPU?

**Comment by Jalle19** [comment]
```
~ # addr2line -a -e /sbin/minisatip
0x418f8a
0x00418f8a
/build/apps/minisatip/src/utils.c:880
0x4190f6
0x004190f6
/build/apps/minisatip/src/utils.c:963
```

Could be that my build is not identical though so take these with a grain of salt

**Comment by Jalle19** [comment]
@estimadarocha https://github.com/Jalle19/satip-axe/releases/tag/build42-rc2 can you try to reproduce any crashes you've been experiencing lately with this build? It has the `addr2line` utility available so it should actually show where the crash happens.

The build contains minisatip 1.2.12, if you can't get it to crash on that version let me know and I'll make you one with the latest release (compiled _without_ PMT=0, because I think that's what making it crash).

**Comment by estimadarocha** [comment]
Yes i can. But i believe they wont happen with that version

**Comment by catalinii** [comment]
addr2line in base image will speed the troubleshooting. @estimadarocha can you upgrade to that image?

**Comment by Jalle19** [comment]
@roisiano can you reproduce the issue with https://github.com/Jalle19/satip-axe/releases/tag/build24-rc3 and a copy of minisatip from build 23?

**Comment by estimadarocha** [comment]
@Jalle19 testing [Build 24 (rc3)](https://github.com/Jalle19/satip-axe/releases/tag/build24-rc3)

**Comment by estimadarocha** [comment]
> @estimadarocha https://github.com/Jalle19/satip-axe/releases/tag/build42-rc2 can you try to reproduce any crashes you've been experiencing lately with this build? It has the `addr2line` utility available so it should actually show where the crash happens.
> 
> The build contains minisatip 1.2.12, if you can't get it to crash on that version let me know and I'll make you one with the latest release (compiled _without_ PMT=0, because I think that's what making it crash).

no... with 1.2.12 it works!

**Comment by Jalle19** [comment]
[satip-axe-202212201649-24-rc3-test1.zip](https://github.com/Jalle19/satip-axe/files/10270453/satip-axe-202212201649-24-rc3-test1.zip)

@estimadarocha can you try this build? It has minisatip v1.2.49.

**Comment by roisiano** [comment]
> @roisiano can you reproduce the issue with https://github.com/Jalle19/satip-axe/releases/tag/build24-rc3 and a copy of minisatip from build 23?

Sorry, I won't be at the place where the Digibit R1 is for a few weeks. If I can, I'll try when I'm there.
Could you tell me how to see and upload the logs?

Kind regards

**Comment by Jalle19** [comment]
@roisiano @estimadarocha have you had a chance to tets the build I linked here (24-rc3-test1)?

**Comment by estimadarocha** [comment]
@Jalle19 will test today!

**Comment by catalinii** [comment]
The current master should work now on axe.

**Comment by Jalle19** [comment]
I'll make a new build then 👍

**Comment by estimadarocha** [comment]
@Jalle19 @catalinii thanks

**Comment by roisiano** [comment]
Hello:

Finally, I could test the new releases with Digibit R1.

**Confirmed! 24 (rc3)
https://github.com/Jalle19/satip-axe/releases/tag/build24-rc3 

Seems to work perfectly for me with different channels of the same transponder. I tested until 5 channels from same TP, and 2+3 from 2 different TPs.**

All OK.

Thanks again and kind regards

**Comment by roisiano** [comment]
> @roisiano @estimadarocha have you had a chance to tets the build I linked here (24-rc3-test1)?

https://github.com/Jalle19/satip-axe/releases/tag/build24-rc3 

Works OK with several channels of the same TP.

Kind regards

**Comment by Jalle19** [comment]
https://github.com/Jalle19/satip-axe/releases/tag/build24
